models/bert_feature_extractor_model.py

```diff
@@ -28,31 +31,28 @@ def average_axis(axis: int, tensor: Tensor) -> Tensor:
 
 class BERTAsFeatureExtractorEncoder(nn.Module):
     def __init__(
-        self,
-        bert_version: BERTVersion,
-        hidden_size: int = None,
-        bert_reducer: Callable[[Tensor], Tensor] = average_layers_and_tokens
+            self,
+            bert_version: BERTVersion,
+            hidden_size: int = None,
+            bert_reducer: Callable[[Tensor], Tensor] = average_layers_and_tokens,
+            device=None
     ):
         super().__init__()
+        self.device = device
+        if not self.device:
+            self.device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
 
         self.bert_version = bert_version.value
         self.bert_reducer = bert_reducer
         self.config = AutoConfig.from_pretrained(self.bert_version, output_hidden_states=True, return_dict=True)
         self.tokenizer = AutoTokenizer.from_pretrained(self.bert_version, config=self.config)
         self.bert = AutoModel.from_pretrained(self.bert_version, config=self.config)
+        self.bert.to('cuda')
```

Comment:
This should be conditional for sure. Perhaps also separate from the model itself?

Reply:
You could parameterize the device like I did:

https://github.com/e2e-retrieval-in-continuous-space/e2e-retrieval/blob/main/models/simple_embedding_model.py